MyApplication/app/src/main/java/com/scottlindley/listview_listadapters_lab/MainActivity.java

```diff
@@ -0,0 +1,154 @@
+
+public class MainActivity extends AppCompatActivity {
+
+    private ListView mListView;
+    private BaseAdapter mBaseAdapter;
+    private ArrayList<Book> mBooks;
+    private Button mSortButton;
+    private Spinner mSpinner;
+    private ArrayList<String> mSortingOptions;
+    private ArrayAdapter<String> mArrayAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mSortButton = (Button)findViewById(R.id.sort_button);
+
+        initializeBooksArray();
+
+        initializeSpinner();
+
+        initializeListView();
+
+        setListViewClick();
+
+        setSortButtonClick();
+
+
+    }
+
+    public void initializeBooksArray(){
+        mBooks = new ArrayList<>();
+        mBooks.add(new Book("The Stand", "Stephen King", 1234));
+        mBooks.add(new Book("The Hobbit", "JRR Tolkein", 321));
+        mBooks.add(new Book("Gone Girl", "Gillian Flynn", 432));
+        mBooks.add(new Book("The Lorax", "Dr. Seuss", 45));
+    }
+
+    public void initializeSpinner(){
+        mSpinner = (Spinner)findViewById(R.id.drop_down);
+        mSortingOptions = new ArrayList<>();
+        mSortingOptions.add("Title");
+        mSortingOptions.add("Author");
+        mSortingOptions.add("Pages");
+        mArrayAdapter = new ArrayAdapter<>(MainActivity.this, android.R.layout.simple_spinner_dropdown_item, mSortingOptions);
+        mSpinner.setAdapter(mArrayAdapter);
+        mArrayAdapter.notifyDataSetChanged();
+    }
+
+    public void initializeListView(){
+        mListView = (ListView)findViewById(R.id.book_list);
+        mBaseAdapter = new BaseAdapter() {
+            @Override
+            public int getCount() {
+                return mBooks.size();
+            }
+
+            @Override
+            public Object getItem(int i) {
+                return mBooks.get(i);
+            }
+
+            @Override
+            public long getItemId(int i) {
+                return i;
+            }
+
+            @Override
+            public View getView(int i, View view, ViewGroup viewGroup) {
+                if(view == null){
+                    LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
+                    view = inflater.inflate(R.layout.list_view_element, null);
+                }
+                TextView title = (TextView)view.findViewById(R.id.title);
+                title.setText("Title: "+mBooks.get(i).getTitle());
+                TextView author = (TextView)view.findViewById(R.id.author);
+                author.setText("Author: "+mBooks.get(i).getAuthor());
+                TextView pages = (TextView)view.findViewById(R.id.numPages);
+                pages.setText("Pages: "+Integer.toString(mBooks.get(i).getNumberPages()));
+                return view;
+            }
+        };
+        mListView.setAdapter(mBaseAdapter);
+        mBaseAdapter.notifyDataSetChanged();
```

Comment:
No need to call `notifyDataSetChanged()` after setting the adapter - it'll go through and set all the views automatically when `setAdapter()` is called